full-moon-derive/src/symbols.rs

```diff
@@ -126,20 +113,25 @@ pub fn parse(input: proc_macro::TokenStream) -> proc_macro::TokenStream {
             }
         }
 
-        fn parse_symbol(code: &str) -> IResult<&str, Symbol> {
-            use ::nom::{
-                branch::alt,
-                bytes::complete::{tag},
-                character::complete::{alphanumeric1},
-                combinator::{not, value},
-                sequence::terminated,
-            };
-            let mut combinator = alt((
-                #(
-                    #splits,
-                )*
-            ));
-            combinator(code)
+        fn parse_keyword(identifier: &str) -> Option<Symbol> {
+            Some(match identifier {
+                #(#identifier_match)*
+                _ => return None,
+            })
+        }
+
+        trait ParseSymbol<'input> {
+            fn parse_symbol(self, pos: usize) -> peg::RuleResult<Symbol>;
+        }
+        impl<'input> ParseSymbol<'input> for &'input str {
+            fn parse_symbol(self: Self, pos:usize) -> peg::RuleResult<Symbol> {
+                for (symbol, string) in &[#(#operator_array,)*] {
+                    if self[pos..].starts_with(string) {
+                        return peg::RuleResult::Matched(pos+string.len(), *symbol);
```

Comment:
Rustfmt doesn't parse these, but they should still be aptly formatted.

```suggestion
                        return peg::RuleResult::Matched(pos + string.len(), *symbol);
```